src/expr/datatype.cpp

```diff
@@ -1350,7 +723,26 @@ Type DatatypeConstructorArg::getRangeType() const {
 
 bool DatatypeConstructorArg::isUnresolvedSelf() const
 {
-  return d_selector.isNull() && d_name.size() == d_name.find('\0') + 1;
+  std::string name = getName();
+  return d_selector.isNull() && name.size() == name.find('\0') + 1;
```

Comment:
Is the second conjunct of this condition still relevant?

Reply:
I don't think so, I'll remove.